### PR TITLE
Initial pass-through for some light cleanup

### PR DESCRIPTION
**Pull request by kevans91**
None of this entailed any functional changes; generally just hiding implementation details that don't need to be global and moving declarations into headers to make it less error-prone.

There were duplicate syscall.c -- are both needed? I've removed one because gmake's kind of dumb and picks the first one it finds in a PATH search. The version in Src/ is slightly better, as it uses weak symbols for _read/_write so one could theoretically override those.

### CODE REVIEW

**Comment by Ownasaurus** [comment]
Thank you very much. I hope to look into this by the end of the weekend.

File: Src/stm32f4xx_it.c [hunk 1]
```diff
@@ -88,23 +88,23 @@ const uint8_t V2_CLOCK_LOW_A = 31;
 
 /* Private macro -------------------------------------------------------------*/
 /* USER CODE BEGIN PM */
-#define WAIT_4_CYCLES asm("ADD     R1, R2, #0\nADD     R1, R2, #0\nADD     R1, R2, #0\nADD     R1, R2, #0")
+#define WAIT_4_CYCLES __asm("ADD     R1, R2, #0\nADD     R1, R2, #0\nADD     R1, R2, #0\nADD     R1, R2, #0")
```
Comment (Ownasaurus) [inline Src/stm32f4xx_it.c]: Why __asm() vs asm()? Please explain. Thanks :)

Comment (kevans91) [reply Src/stm32f4xx_it.c]: 'asm' is the GNU-extension spelling of '__asm' -- the former isn't available when attempting to compile in C99/C11 mode (the default is typically C11 + GNU extensions in GCC-land), so this is effectively a low-cost move away from GNU extensions where the C-specified identifier isn't much less readable.

**Comment by Ownasaurus** [comment]
The code has been tested and confirmed functional using our most demanding payload. Merging to master. Great job!

**Comment by kevans91** [comment]
Thanks! Can you shoot me some details of your demanding payload setup? I've got a new soldering tip enroute and some more flux to finish up my board here in the next week or so. =-)

**Pull request merged**